main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"

```

Comment:
Can you refactor to use `gin` instead? It's written very clearly in step 5 of the task description. Also, make [this project modular](https://github.com/golang/go/wiki/Modules#quick-start) and include `go.mod` changes in this PR.